src/terminal/input/terminalInput.cpp

```diff
@@ -466,6 +468,44 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent) const
     return fKeyHandled;
 }
 
+bool TerminalInput::HandleChar(const char16_t ch)
+{
+    if (ch == UNICODE_BACKSPACE || ch == UNICODE_DEL)
+    {
+        return false;
+    }
+    else if (Utf16Parser::IsLeadingSurrogate(ch))
+    {
+        if (_leadingSurrogate.has_value())
+        {
+            // we already were storing a leading surrogate but we got another one. Go ahead and send the
+            // saved surrogate piece and save the new one
+            wchar_t buffer[32];
+            swprintf_s(buffer, L"%I32u", _leadingSurrogate.value());
+            _SendInputSequence(buffer);
+        }
+        // save the leading portion of a surrogate pair so that they can be sent at the same time
+        _leadingSurrogate.emplace(ch);
+    }
+    else if (_leadingSurrogate.has_value())
+    {
+        std::wstring wstr;
+        wstr.reserve(2);
+        wstr.push_back(_leadingSurrogate.value());
+        wstr.push_back(ch);
+        _leadingSurrogate.reset();
+
+        _SendInputSequence(wstr.c_str());

```

Comment:
Any chance we can make another entrypoint to _SendInputSequence that takes a `std::wstring_view` so we can save the counting step for this one and the one below where we already know the length? And then leave a second version of it that accepts the z-terminated PCWSTR (and forwards to the wstring_view one) so we only count the string when we didn't already know its length?